parsing/parsetree.mli

```diff
@@ -71,90 +68,108 @@ and attributes = attribute list
 
 and payload =
   | PStr of structure
-  | PSig of signature (* : SIG *)
-  | PTyp of core_type  (* : T *)
-  | PPat of pattern * expression option  (* ? P  or  ? P when E *)
+  | PSig of signature
+      (** [: SIG] in an attribute or an extension point *)
+  | PTyp of core_type
+      (** [: T] in an attribute or an extension point *)
+  | PPat of pattern * expression option
+      (** [? P]  or  [? P when E], in an attribute or an extension point *)
 
 (** {1 Core language} *)
 
-(* Type expressions *)
+(** {2 Type expressions} *)
 
 and core_type =
     {
      ptyp_desc: core_type_desc;
      ptyp_loc: Location.t;
      ptyp_loc_stack: location_stack;
-     ptyp_attributes: attributes; (* ... [@id1] [@id2] *)
+     ptyp_attributes: attributes;  (** [... [\@id1] [\@id2]] *)
     }
 
 and core_type_desc =
   | Ptyp_any
-        (*  _ *)
+        (** [_] *)
   | Ptyp_var of string
-        (* 'a *)
+        (** A type var such as ['a] *)
   | Ptyp_arrow of arg_label * core_type * core_type
-        (* T1 -> T2       Simple
-           ~l:T1 -> T2    Labelled
-           ?l:T1 -> T2    Optional
+        (** [Ptyp_arrow(lbl, T1, T2)] represents:
+            - [T1 -> T2]    when [lbl] is  {{!Asttypes.Nolabel}[Nolabel]},
```

Comment:
ocamldoc needs the type name in order to resolve the link `{!Asttypes.arg_label.Nolabel}` (same for all linked fields and constructors)